### PR TITLE
Generate flx files in the gen directory.

### PR DESCRIPTION
**Pull request by pylaligand**
This allows tools to locate them more easily.
A copy step was added to make the files available at their previous location as well.

### CODE REVIEW

File: build/flutter_app.gni [hunk 1]
```diff
@@ -185,6 +187,20 @@ template("flutter_app") {
...
+      "$root_out_dir/$bundle_name",
+    ]
+
+    deps = [
+      ":$package_target_name",
+    ]
+  }
 }
 
 template("flutter_aot_app") {
```
Comment (pylaligand) [inline build/flutter_app.gni]: Should I add this to flutter_aot_app as well? What do you guys think of putting that template in its own file?

**Comment by yyoon** [comment]
@pylaligand 

Seems like this would create the .flx under gen and then copy it to the root_out_dir again, correct?
But then the `user.bootfs.manifest` would still point to the binaries under `root_out_dir` and it wouldn't particularly help the fuchsia_reload tool.

To make this work, I think we should be just creating the .flx under gen and then edit all the modules under //packages/gn to point to the new .flx paths. What do you think?

**Comment by pylaligand** [comment]
It seems I misunderstood how the hot reload tool worked. Let me take another look.

**Comment by pylaligand** [comment]
This change is much more simple now as the heavy-lifting is happening in the Fuchsia tree.

**Pull request merged**